lib/Service/RoomService.php

```diff
@@ -362,6 +366,54 @@ public function setAvatar(Room $room, string $avatar): bool {
 		return true;
 	}
 
+	public function startRecording(Room $room, $status): bool {
+		$availableRecordingTypes = [Room::RECORDING_VIDEO, Room::RECORDING_AUDIO];
+		if (!in_array($status, $availableRecordingTypes)) {
+			throw new InvalidArgumentException('status');
+		}
+		if ($room->getCallRecording() !== 0) {
+			throw new InvalidArgumentException('room');
+		}
+		return $this->setCallRecording($room, $status);
+	}
+
+	public function stopRecording(Room $room): bool {
+		if ($room->getCallRecording() === 0) {
+			throw new InvalidArgumentException('room');
+		}
+		return $this->setCallRecording($room);
+	}
+
+	/**
+	 * @param Room $room
+	 * @param integer $status 0 none|1 video|2 audio
+	 */
+	public function setCallRecording(Room $room, int $status = Room::RECORDING_STOP): bool {
+		if (!$this->config->isRecordingEnabled()) {
+			throw new InvalidArgumentException('config');
+		}
+
+		$availableRecordingStatus = [Room::RECORDING_STOP, Room::RECORDING_VIDEO, Room::RECORDING_AUDIO];
+		if (!in_array($status, $availableRecordingStatus)) {
+			throw new InvalidArgumentException('status');
+		}
+
+		$oldStatus = $room->getCallRecording();
+		$event = new ModifyRoomEvent($room, 'callRecording', $status, $oldStatus);
+		$this->dispatcher->dispatch(Room::EVENT_BEFORE_SET_CALL_RECORDING, $event);
+
+		$update = $this->db->getQueryBuilder();
+		$update->update('talk_rooms')
+			->set('call_recording', $update->createNamedParameter($status, IQueryBuilder::PARAM_INT))
+			->where($update->expr()->eq('id', $update->createNamedParameter($room->getId(), IQueryBuilder::PARAM_INT)));
+		$update->executeStatement();
+
+		$room->setCallRecording($status);
+
+		$this->dispatcher->dispatch(Room::EVENT_AFTER_SET_CALL_RECORDING, $event);
+		return true;
```

Comment:
```suggestion
```

Since in error cases it throws, no need to return anything on success.

That is our new service pattern.